### PR TITLE
Fix CR problems in tests and use printf instead echo

### PR DESCRIPTION
**Pull request by franiglesias**
With this PR we fix some problems with the CR in test files that was causing problems to compare expected and actual output.

Also switched to use printf instead of echo to better compatibility.

Note that currently not all tests are passing expecting to better design the output of the command.

### CODE REVIEW

**Comment by pkristian** [comment]
So now are tests white-space sensitive? right?

What problems with compability you ran into?

**Comment by franiglesias** [comment]
I was having problems running tests in my current installation (Mac OS X):

```
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin17)
Copyright (C) 2007 Free Software Foundation, Inc.
```
- Escape codes didn't work. Switching to printf did the trick
- Some tests didn't pass du to files with CR/LF line endings. Changed to LF

I will need some time to test that under other conditions

**Comment by pkristian** [comment]
and oh yes, i am runing it on `git-bash.exe` which is based on `MINGW64`. FYI :)

**Comment by franiglesias** [comment]
major test failing is related to the .semver thing (which was no present when you wrote the tests. Now that I make them run, I think I will better suited to improve things), so I'm going to fix it first and perform some cosmetic changes in the outputs.

I prefer spaces over tabs, yes. :-)

**Comment by pkristian** [comment]
yep, i al looking forward to you fixing tests.. please also comment my code comments

File: testCases/testAll.sh [hunk 1]
```diff
@@ -12,53 +12,53 @@ do
...
+	printf "### ";
+	printf "\e[0m";
+	printf "${testCase/%?}";
 
 	expected="$(cat $testCase/expected)"
 
 	actual="$(bash $testCase/command.sh)"
 
-	if [ "$expected" == "$actual" ]
+	if [ "$expected" = "$actual" ]
```
[the remark this answers is not in the document]
Comment (franiglesias) [reply testCases/testAll.sh]: Same thing

File: testCases/help/command.sh [hunk 2]
```diff
@@ -1,2 +1 @@
-bash ../versiontag \
-    help
+bash ../versiontag help
```
[the remark this answers is not in the document]
Comment (franiglesias) [reply testCases/help/command.sh]: Same

File: testCases/defaultSetup/command.sh [hunk 3]
```diff
@@ -1 +1 @@
-bash ../versiontag
+bash ../versiontag
```
[the remark this answers is not in the document]
Comment (franiglesias) [reply testCases/defaultSetup/command.sh]: Yeah. I changed things like this looking for a reason to explain how it was possible that no test was passing. I'll fix it.

**Comment by franiglesias** [comment]
Sorry, I forgot to click the send button 😂

**Comment by pkristian** [comment]
good, when you will send updated commit, i will review it and hopefully all will be nice
 :)

**Comment by franiglesias** [comment]
I guess I need some sleep 💤

**Pull request merged**